lib/msal-common/src/authority/RegionDiscovery.ts

```diff
@@ -38,19 +39,27 @@ export class RegionDiscovery {
                 if (response.status === ResponseCodes.httpBadRequest) {
                     const latestIMDSVersion = await this.getCurrentVersion();
                     if (!latestIMDSVersion) {
+                        regionDiscoveryMetadata.region_source = RegionDiscoverySources.FAILED_AUTO_DETECTION;
                         return null;
                     }
 
                     const response = await this.getRegionFromIMDS(latestIMDSVersion);
                     if (response.status === ResponseCodes.httpSuccess) {
                         autodetectedRegionName = response.body;
                     }
-                } 
+                }
+                
+                if (autodetectedRegionName) regionDiscoveryMetadata.region_source = RegionDiscoverySources.IMDS;
```

Comment:
Get rid of this with the above statement

Reply:
The block is a little fuzzy. We can sync up offline to resolve this.

Reply:
I have removed the above statement and refactored the implementation as advised.